tool-openssl/x509_test.cc

```diff
@@ -113,6 +155,31 @@ TEST_F(X509Test, X509ToolModulusTest) {
   ASSERT_TRUE(result);
 }
 
+// Test -subject
```

Comment:
What do we gain from these tests? What if `X509Tool` was a single line function that just returned true, would these tests notice?

Reply:
These tests wouldn't, but the openssl comparison tests would. These tests are simply verifying that we support the advertised options and are able to parse them correctly. I guess the openssl comparison tests also verify this. I think we could possibly make this more efficient by having an array of options and checking them in one function. I can tackle this in a follow up PR along with the refactor for comparison tests suggested below.